whisper/whisperv6/filter.go

```diff
@@ -35,23 +35,29 @@ type Filter struct {
 	PoW        float64           // Proof of work as described in the Whisper spec
 	AllowP2P   bool              // Indicates whether this filter is interested in direct peer-to-peer messages
 	SymKeyHash common.Hash       // The Keccak256Hash of the symmetric key, needed for optimization
+	id         string            // unique identifier
 
 	Messages map[common.Hash]*ReceivedMessage
 	mutex    sync.RWMutex
 }
 
 // Filters represents a collection of filters
 type Filters struct {
-	watchers map[string]*Filter
-	whisper  *Whisper
-	mutex    sync.RWMutex
+	watchers         map[string]*Filter
```

Comment:
Do we really need watchers at all? The only mention I can find is in `Uninstall` and I believe that it could be replaced with a check to `allTopicsMatchers` and a call to `removeFromTopicsMatchers`.

Reply:
they are used for the lookup (finding filter by id)